include/boost/geometry/algorithms/detail/envelope/range.hpp

```diff
@@ -94,16 +99,20 @@ struct envelope_multi_range
                              Box& mbr,
                              Strategy const& strategy)
     {
-        apply(boost::begin(multirange), boost::end(multirange), mbr, strategy);
+        using range_t = typename boost::range_value<MultiRange>::type;
+        using strategy_t = decltype(strategy.envelope(std::declval<range_t>(), mbr));
+        using state_t = typename strategy_t::template multi_state<Box>;
```

Comment:
So this new approach uses C++14 - OK for me.

Reply:
Yes, though this particular part is C++11.

I think the only thing from C++14 I used is `auto` return type in umbrella strategies.

Another thing I could use is type_traits's template aliases, so e.g. `std::enable_if_t` instead of `std::enable_if`, but for now I'm using C++11 versions.

Reply:
I'm fine with either - but is there a particular reason to use the C++11 version here?